cirq/google/calibration/workflow.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+
+from cirq.google.calibration.phased_fsim import (
+    PhasedFSimCalibrationRequest,
+    PhasedFSimCalibrationResult,
+)
+from cirq.google.engine import Engine
+from cirq.google.serializable_gate_set import SerializableGateSet
+
+
+def run_characterizations(
+    calibrations: List[PhasedFSimCalibrationRequest],
+    engine: Engine,
+    processor_id: str,
+    gate_set: SerializableGateSet,
+) -> List[PhasedFSimCalibrationResult]:
```

Comment:
Do we have an object that contains `(engine, processor_id, gate_set)` that we can use instead of having the user have to specify all of these? `QuantumEngineSampler` solves this for running programs but I don't know if it has a `run_calibration` method. We should add this method and then accept an `EngineSampler`.

In general, I think we should be encouraging the use of the `cirq.google.get_engine_sampler` family of functions. For sampling, it lets you sub in a simulator (which is nice) but even for things like this where there isn't a comparable simulator method (side note: maybe there should be a simulator path for `run_calibration` for users to test that it's working) it is useful to group these three objects together. Furthermore, the `get_engine_xx` functions can grab the `project_id` from environment variables, making it easier to version control and distribute demos without a hardcodded project id

Reply:
I agree with all what you just said but can that be done in a follow-up after we have a basic features working?

As for simulator for run_calibrations I have already introduced it as one of the follow-up PRs. It is a necessary thing to be able to do unit-testing of the calibration functionality. It might also have other uses in the future.

Reply:
sounds good